### PR TITLE
pass the network driver to the constructor

### PR DESCRIPTION
**Pull request by adecaro**
This PR simplify the constructors used to create a new channel. It also replaces certain dependencies with interfaces and not structs.

This PR is tested against FTS in https://github.com/hyperledger-labs/fabric-token-sdk/pull/804

### CODE REVIEW

File: ci/fsc-fts-update.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
```
Comment (mbrandenburger) [inline ci/fsc-fts-update.sh:1]: I believe this script belongs to the fts repo

File: platform/fabric/sdk/dig/generic/providers.go [hunk 2]
```diff
@@ -27,6 +28,7 @@ import (
 	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/kvs"
 	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/metrics"
 	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/pkg/errors"
```
Comment (mbrandenburger) [inline platform/fabric/sdk/dig/generic/providers.go:31]:
```suggestion
	"github.com/hyperledger-labs/fabric-smart-client/pkg/utils/errors"
```

**Review by mbrandenburger — CHANGES_REQUESTED** [review]
Overall looks good to me. Just two comments.

File: ci/fsc-fts-update.sh (hunk 1 above)
Comment (adecaro) [reply ci/fsc-fts-update.sh:1]: I really meant to have a script that updates FTS with the current FSC commit.

Comment (mbrandenburger) [reply ci/fsc-fts-update.sh:1]: I understand. However, I think this script should move to the FTS repo as it creates a PR on the FTS repo.

Such a script could support multiple strategies:
- Get the commit hash from any local git clone / working branch.
- Get the lastest commit hash from main
- Provide the commit hash as a parameter to the script and update all go.mod files inside the project.

I believe the right way would be to use dependabot and schedule a daily check if the main branch on FSC has something new. As dependabot only "detects" versioned updated (not individual commits) it won't work like that. You can create a GH action that is fired once per day to check if there is a new commit on FSC main branch and then create a PR.

Comment (adecaro) [reply ci/fsc-fts-update.sh:1]: okay, I'll remove the script.

**Review by mbrandenburger — APPROVED** [review]

File: platform/fabric/core/generic/channelprovider.go [hunk 3]
```diff
@@ -176,36 +186,35 @@ func (p *provider) NewChannel(nw driver.FabricNetworkService, channelName string
...
-		nw.TransactionManager(),
 	)
+	if err != nil {
+		return nil, errors.WithMessagef(err, "failed creating ledger for channel [%s]", channelName)
+	}
 
-	committerService := p.newCommitter(
-		nw.ConfigService(),
+	committerService, err := p.newCommitter(
+		nw,
```
Comment (alexandrosfilios) [inline platform/fabric/core/generic/channelprovider.go:199]: Why do we pass the whole `nw`? We just want a few services, right?

File: platform/fabric/core/generic/ordering/ordering.go [hunk 4]
```diff
@@ -133,6 +137,16 @@ func (o *Service) SetConsensusType(consensusType ConsensusType) error {
 	return nil
 }
 
+func (f *Service) SetConfigOrderers(o channelconfig.Orderer, orderers []*grpc.ConnectionConfig) error {
```
Comment (alexandrosfilios) [inline platform/fabric/core/generic/ordering/ordering.go:140]: We pass `o` that already contains `Consenters` and `orderers`. Do we need both? We could only pass ordererType.

File: platform/fabric/core/generic/channelprovider.go [hunk 5]
```diff
@@ -28,41 +28,41 @@ import (
...
+type VaultConstructor = func(
+	configService driver.ConfigService,
+	channel string,
+	drivers []driver2.NamedDriver,
+	metricsProvider metrics.Provider,
+	tracerProvider trace.TracerProvider,
+) (*vault.Vault, driver.TXIDStore, error)
 type LedgerConstructor func(
 	channelName string,
+	nw driver.FabricNetworkService,
```
Comment (alexandrosfilios) [inline platform/fabric/core/generic/channelprovider.go:40]: I don't know if it's a good idea to replace the actual dependencies by a struct that contains these and many more. This creates a tighter coupling and looser cohesion.

**Review by alexandrosfilios — CHANGES_REQUESTED** [review]

File: platform/fabric/core/generic/ordering/ordering.go (hunk 4 above)
Comment (adecaro) [reply platform/fabric/core/generic/ordering/ordering.go:140]: good point, we can simplify this.

**Review by alexandrosfilios — COMMENTED** [review]
LGTM

**Review by alexandrosfilios — APPROVED** [review]
LGTM

**Pull request merged**